backend/test/v2/test_utils.go

```diff
@@ -129,8 +129,35 @@ func ListAllRuns(client *api_server.RunClient, namespace string) ([]*run_model.V
 func DeleteAllRuns(client *api_server.RunClient, namespace string, t *testing.T) {
 	runs, _, _, err := ListAllRuns(client, namespace)
 	assert.Nil(t, err)
-	for _, r := range runs {
-		assert.Nil(t, client.Delete(&run_params.RunServiceDeleteRunParams{RunID: r.RunID}))
+	for _, run := range runs {
+		err := client.Delete(&run_params.RunServiceDeleteRunParams{RunID: run.RunID})
+		// In some cases, the run may have been deleted by other tests.
+		// We can ignore the not found error.
+		if err != nil {
+			assert.Contains(t, err.Error(), "not found")
+		}
+	}
+	// Wait for runs to be deleted.
+	// Increased timeout for local dev environment where cleanup can be slow
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(60*time.Second))
+	defer cancel()
+
+	for {
+		select {
+		case <-ctx.Done():
+			remainingRuns, _, _, _ := ListAllRuns(client, namespace)
+			var remainingRunNames []string
+			for _, r := range remainingRuns {
+				remainingRunNames = append(remainingRunNames, r.DisplayName)
+			}
+			require.FailNowf(t, "Runs were not deleted after 15 seconds", "Remaining runs: %v", remainingRunNames)
```

Comment:
timeout is 60 not 15, I would suggest having this value be parameterized rather than hardcoded to prevent such discrepancies